src/rpdk/core/contract/resource_client.py

```diff
@@ -215,28 +215,67 @@ def get_metadata(self):
                 and properties[prop]["insertionOrder"] == "false"
             }
 
-    def transform_model(self, input_model, output_model):
+    def transform_model(self, input_model, output_model):  # pylint: disable=R0914
         if self.property_transform_keys:
             self.check_npm()
-        for prop in self.property_transform_keys:
-            document_input, _path_input, _parent_input = traverse(
-                input_model, list(prop)[1:]
-            )
-            document_output, _path_output, _parent_output = traverse(
-                output_model, list(prop)[1:]
-            )
-            if document_input != document_output:
-                transformed_property = self.transform(prop, input_model)
-                self.update_transformed_property(
-                    prop, transformed_property, input_model
+        for prop in self.property_transform_keys:  # pylint: disable=R1702
+            try:
+                document_input, _path_input, _parent_input = traverse(
+                    input_model, list(prop)[1:]
                 )
+                document_output, _path_output, _parent_output = traverse(
+                    output_model, list(prop)[1:]
+                )
+                if not self.compare(document_input, document_output):
+                    path = "/" + "/".join(prop)
+                    property_transform_value = self.property_transform[path].replace(
+                        '"', '\\"'
+                    )
+                    if "OR" in property_transform_value:
+                        transform_functions = property_transform_value.split("OR")
```

Comment:
OR (in upper case) is not supported in jsonata. It makes sense that I append $ for OR to denote a function. As for the space, I don't think it will matter. In this case trailing or leading spaces will not matter for jsonata, I would add a trim function to avoid any issues though. 
Please let me know if I missing the point.

Reply:
I phrased that a bit unclear. consider this (bit contrived schema fragment):

```
"propertyTransform": {
     "/properties/a": "$join([a, 'ORGANIZATION'])"
}
```
`property_transform_value.split("OR")` will return `['$join([a, "', 'GANIZATION"])']`.

The safest way to avoid that is to not split the string at all:
```
# this could be like IAM policies, either a string if there is only a single element or a list for OR behaviour
"propertyTransform": {
     "/properties/a": [
        "$join([a, 'ORGANIZATION'])",
        "$join([a, 'something else'])"
}
```

The mitigation I proposed to also look at the `$` being behind the OR  to reduce the likelihood of the separator being part of a word is probably less useful than I originally thought. There is a chance of both parts starting with `$`, but it is not guaranteed

Reply:
Edit: That might be more of a schema discussion than a contract test discussion, but since I haven't seen the OR behaviour documented, it might make sense to have that discussion anyway